server/lib/util.py

```diff
@@ -749,67 +749,12 @@ def get_series_dates_from_entities(entities: List[str], variables: List[str]):
 
   result = {
       'datesByVariable': dates_by_variable,
-      'facets': obs_series_response['facets']
+      'facets': obs_series_response.get('facets', {})
   }
   return result
 
 
-def fetch_highest_coverage(parent_entity: str,
-                           child_type: str,
-                           variables: List[str],
-                           all_facets: bool,
-                           facet_ids: List[str] = None):
-  """
-  Fetches the latest available data with the best coverage for the given a
-  parent entity, child type, variables, and facets. If multiple variables are
-  passed in, selects dates with highest coverage independently for each
-  variable.
-
-  Response format:
-  {
-    "facets": {
-      <facet_id>: {<facet object>}
-    },
-    "data": {
-      <var_dcid>: {
-        <entity_dcid>: {
-          <observation point(s)>
-        }
-      }
-    }
-  }
-  """
-  MAX_DATES_TO_CHECK = 5
-  MAX_YEARS_TO_CHECK = 5
-  point_responses = []
-  series_dates_response = dc.get_series_dates(parent_entity, child_type,
-                                              variables)
-  facet_ids_set = set(facet_ids or [])
-  for observation_entity_counts_by_date in series_dates_response[
-      'datesByVariable']:
-    # Each observation_entity_counts_by_date contains the observation counts by date
-    variable = observation_entity_counts_by_date['variable']
-    best_coverage_date = _get_highest_coverage_date(
-        observation_entity_counts_by_date=observation_entity_counts_by_date,
-        facet_ids=facet_ids_set,
-        max_dates_to_check=MAX_DATES_TO_CHECK,
-        max_years_to_check=MAX_YEARS_TO_CHECK)
-    if not best_coverage_date:
-      # No best coverage date means we couldn't find any variable observations
-      # Add a blank point response in this case
-      point_responses.append({'data': {variable: {}}})
-      continue
-    point_responses.append(
-        fetch.point_within_core(parent_entity, child_type, [variable],
-                                best_coverage_date, all_facets, facet_ids))
-  combined_point_response = {"facets": {}, "data": {}}
-  for point_response in point_responses:
-    combined_point_response["facets"].update(point_response.get("facets", {}))
-    combined_point_response["data"].update(point_response.get("data", {}))
-  return combined_point_response
-
-
-def _get_highest_coverage_date(observation_entity_counts_by_date,
+def _get_highest_coverage_date(observation_dates_by_variable,
```

Comment:
nit: please update docstring to describe `observation_dates_by_variable` in "Args:" instead of `observation_entity_counts_by_date`

Reply:
done!